R/animint.R

```diff
@@ -22,11 +22,34 @@ parsePlot <- function(meta){
       plot.meta$scales[[sc$aesthetics]] <- makeScale(sc)
     }
   }
+  
+  ## Export axis specification as a combination of breaks and
+  ## labels, on the relevant axis scale (i.e. so that it can
+  ## be passed into d3 on the x axis scale instead of on the
+  ## grid 0-1 scale). This allows transformations to be used
+  ## out of the box, with no additional d3 coding.
+  theme.pars <- ggplot2:::plot_theme(meta$plot)
+  
+  ## No legend if theme(legend.postion="none").
+  plot.meta$legend <- if(theme.pars$legend.position != "none"){
+    getLegendList(meta$built)
+  }
+  
+  ## save each layer
   for(layer.i in seq_along(meta$plot$layers)){
     ##cat(sprintf("%4d / %4d layers\n", layer.i, length(meta$plot$layers)))
 
     ## This is the layer from the original ggplot object.
     L <- meta$plot$layers[[layer.i]]
+    
+    ## If any legends are specified, add showSelected aesthetic
+    for(legend.i in seq_along(plot.meta$legend)) {
+      var_name <- plot.meta$legend[[legend.i]]$vars
+      var <- L$data[[var_name]]
+      if(is.factor(var) | is.character(var)) {
+        L$mapping$showSelectedcolour <- as.symbol(var_name)
+      }
+    }
```

Comment:
This is my initial idea for adding showSelected aesthetics.  @tdhock is this what you had in mind?